src/main/kotlin/io/perfometer/data/csv/CsvParser.kt

```diff
@@ -0,0 +1,61 @@
+package io.perfometer.data.csv
+
+internal class CsvParser(private val csvText: String, private val delimiter: Char = ',') {
+
+    private var position = 0
+
+    fun parse(): List<List<String>> {
+        position = 0
+        val result = mutableListOf<List<String>>()
+        while (notFinished()) {
+            result.add(parseRecord())
+        }
+        return result
+    }
+
+    private fun parseRecord(): List<String> {
+        val result = mutableListOf<String>()
+        while (checkCurrentCharIsNot('\n')) {
```

Comment:
I suggest avoiding negation in method names, like in this example: `checkCurrentCharIsNot`. In my experience such names bring a little more cognitive load for someone reading the code, especially when additionally negated with `!` - 
`!checkCurrentCharIsNot('\n')`

Please rename the method to, e.g. `isChar('\n') and use `!` negation:
```kotlin
while(!isChar('\n')) {
...
}
```

Reply:
At the moment is not as simple because these methods are also checking for the end of input and returning false in both cases if input has ended. I could inline that but it would be a repetition then.
I've changed method names to make it more clear and consistent.